hivelocity/instancesV2_test.go

```diff
@@ -110,90 +100,125 @@ func standardMocks(m *mocks.API) {
-	require.NoError(t, err)
-	require.Equal(t, true, myBool)
-
-	node.Spec.ProviderID = "9999999"
-	myBool, err = i2.InstanceExists(ctx, node)
-	require.Equal(t, false, myBool)
-	require.NoError(t, err)
+	m := mocks.NewAPI(t)
+	i2, node, ctx := newTestData(m)
 
-	node.Spec.ProviderID = "9999999999999999999999999999"
-	myBool, err = i2.InstanceExists(ctx, node)
-	require.Equal(t, false, myBool)
-	require.Equal(t, "failed to convert node.Spec.ProviderID \"9999999999999999999999999999\" to int32", err.Error())
+	tests := []struct {
+		providerId    int64
+		wantBool      bool
+		wantErrString string
+	}{
+		{
+			providerId:    int64(mockDeviceId),
+			wantBool:      true,
+			wantErrString: "",
+		},
+		{
+			providerId:    9999999,
+			wantBool:      false,
+			wantErrString: "",
+		},
+		{
+			providerId:    999999999999999999,
+			wantBool:      false,
+			wantErrString: "GetHivelocityDeviceIdFromNode(node) failed: failed to convert node.Spec.ProviderID \"hivelocity://999999999999999999\" to int32",
+		},
+	}
+	for _, tt := range tests {
+		node.Spec.ProviderID = fmt.Sprintf("hivelocity://%d", tt.providerId)
+		gotBool, gotErr := i2.InstanceExists(ctx, node)
+		msg := fmt.Sprintf("Input: providerId=%+v", tt.providerId)
+		if tt.wantErrString == "" {
+			require.NoError(t, gotErr, msg)
+		} else {
+			require.Error(t, gotErr, msg)
+			require.Equal(t, tt.wantErrString, gotErr.Error(), msg)
+		}
+		require.Equal(t, tt.wantBool, gotBool, tt.providerId, msg)
+	}
 }
 
 func Test_InstanceShutdown(t *testing.T) {
-	i2, m := newHVInstanceV2(t)
-	standardMocks(m)
-	node := newNode()
-	ctx := context.Background()
-	isDown, err := i2.InstanceShutdown(ctx, node)
-	require.False(t, isDown)
-	require.NoError(t, err)
-
-	node.Spec.ProviderID = "9999999"
-	_, err = i2.InstanceShutdown(ctx, node)
-	require.Error(t, err)
-}
-
-func Test_InstanceMetadata(t *testing.T) {
-	i2, m := newHVInstanceV2(t)
-	node := newNode()
-	ctx := context.Background()
-	standardMocks(m)
-	metaData, err := i2.InstanceMetadata(ctx, node)
-	require.NoError(t, err)
-	require.Equal(t, &cloudprovider.InstanceMetadata{
-		ProviderID: "14730",
-		NodeAddresses: []corev1.NodeAddress{
-			{
-				Type:    corev1.NodeAddressType("ExternalIP"),
-				Address: "66.165.243.74",
-			},
+	m := mocks.NewAPI(t)
+	i2, node, ctx := newTestData(m)
+	tests := []struct {
+		providerId    int
+		wantBool      bool
+		wantErrString string
+	}{
+		{
+			providerId:    mockDeviceId,
+			wantBool:      false,
+			wantErrString: "",
 		},
-		Zone:   "LAX2",
-		Region: "LAX2",
-	}, metaData)
-
-	node.Spec.ProviderID = "9999999"
-	metaData, err = i2.InstanceMetadata(ctx, node)
-	require.Error(t, err)
-	require.Nil(t, metaData)
+		{
+			providerId:    9999999,
+			wantBool:      false,
+			wantErrString: "i2.API.GetBareMetalDeviceIdResource(deviceId) failed: no such device",
+		},
+	}
+	for _, tt := range tests {
+		node.Spec.ProviderID = fmt.Sprintf("hivelocity://%d", tt.providerId)
```

Comment:
doesn't it make sense to have a function returning a node with some input parameters?
I don't really like this pattern of defining the node first and then altering it afterward...